nd4j/nd4j-backends/nd4j-backend-impls/nd4j-native/src/main/java/org/nd4j/linalg/cpu/nativecpu/CpuNDArrayFactory.java

```diff
@@ -90,6 +90,55 @@ public void createBlas() {
             System.setProperty(ND4JSystemProperties.ND4J_CPU_LOAD_OPENBLAS_NOLAPACK, "mklml");
         }
 
+        // we'll check hardware support first
+        if (!nativeOps.isMinimalRequirementsMet()) {
+            // this means cpu binary was built for some arch support, we don't have on this box
+
+            val binaryLevel = nativeOps.binaryLevel();
+            val optimalLevel = nativeOps.optimalLevel();
+
+            switch (binaryLevel) {
+                case 3: {
+                        // avx-512 binary
+                        log.error("Trying to use AVX-512 binary without actual CPU support for it");
```

Comment:
It would probably be nicer to use JavaCPP to load the right binaries instead of only giving warnings like that.

Reply:
RIght, that's exactly the plan. We want jcpp to manage that, but in case user explicitly uses, say, avx512 on one box, and tries to run on avx2 box - instead of SIGILL crash he'll see nice message that explains what happens and what to do.